### PR TITLE
Dev/privacy v2 reduce txs size

### PR DESCRIPTION
**Pull request by thanhn-inc**
- Added ToCompactBytes and FromCompactBytes to TxVer2 and TxTokenV2. ToCompactBytes should only be used when storing the data, and must not be used to calculate hashes.
- Each metadata now has to implement the method  `ToCompactBytes` (or use the default method), some have been implemented

### CODE REVIEW

File: metadata/pdexv3/trade.go [hunk 1]
```diff
@@ -188,6 +190,71 @@ func (req *TradeRequest) CalculateSize() uint64 {
...
+					return err
+				}
+				req.Receiver[*tokenID] = *otaReceiver
+			}
+
+			return nil
+		}
+	}
+
+	return fmt.Errorf("not an TradeRequest")
```
Comment (dat-incognito-org) [inline metadata/pdexv3/trade.go:255]: an -> a

File: metadata/parse.go [hunk 2]
```diff
@@ -263,3 +273,29 @@ func ParseMetadata(meta interface{}) (Metadata, error) {
 
 	return md, nil
 }
+
+func MdFromCompactBytes(data []byte) (Metadata, error) {
+	if len(data) < 2 {
```
Comment (dat-incognito-org) [inline metadata/parse.go:278]: should there be an error here ?

File: metadata/parse.go [hunk 3]
```diff
@@ -21,7 +23,15 @@ func ParseMetadata(meta interface{}) (Metadata, error) {
 	}
 	err = json.Unmarshal(metaInBytes, &mtTemp)
 	if err != nil {
-		return nil, err
```
Comment (dat-incognito-org) [inline metadata/parse.go:24]: this function needs simplification

File: metadata/pdexv3/trade.go (hunk 1 above)
Comment (thanhn-inc) [reply metadata/pdexv3/trade.go]: Fixed

File: metadata/parse.go (hunk 2 above)
Comment (thanhn-inc) [reply metadata/parse.go:265]: Fixed

File: metadata/parse.go (hunk 3 above)
Comment (thanhn-inc) [reply metadata/parse.go:24]: Fixed in 3c0822d26477318351a7f68c6be1e31b71547fd1

**Review by dat-incognito-org — APPROVED** [review]

**Comment by 0xankylosaurus** [comment]
Merged into storage reduction code

**Pull request closed without merging**